Public/Src/Cache/ContentStore/Distributed/Sessions/ReadOnlyDistributedContentSession.cs

```diff
@@ -696,25 +738,37 @@ private Task<BoolResult> UpdateContentTrackerWithNewReplicaAsync(Context context
             Context context,
             IReadOnlyList<ContentHash> contentHashes,
             CancellationToken cts,
+            bool onlyCheckGlobalExistence,
             UrgencyHint urgencyHint = UrgencyHint.Nominal)
         {
             var operationContext = new OperationContext(context, cts);
 
-            return Workflows.RunWithFallback(
-                contentHashes,
-                hashes => PinFromContentLocationStoreOriginAsync(operationContext, hashes, cts, GetBulkOrigin.Local, urgencyHint),
-                hashes => PinFromContentLocationStoreOriginAsync(operationContext, hashes, cts, GetBulkOrigin.Global, urgencyHint),
-                result => result.Succeeded);
+            if (onlyCheckGlobalExistence)
+            {
+                return Workflows.RunWithFallback(
+                    contentHashes,
+                    hashes => PinFromContentLocationStoreOriginAsync(operationContext, hashes, cts, GetBulkOrigin.Global, true, urgencyHint),

```

Comment:
In this case you always will be hitting the redis.

You still can look into the local location store here as well...

Lets discuss this.